cloud-watch-to-slack-testing-58006136-c30b-43d1-880c-7b36f860e7c8/deployment/index.js

```diff
@@ -5,54 +5,61 @@
+ *   4. Copy the webhook URL from the setup instructions (e.g. "https://hooks.slack.com/services/abc123").
  *
- *   4. Copy the webhook URL from the setup instructions and use it in the next section.
+ *   5. On the Lambda console window click Edit the environment variable section and environment
+ *      variable "slackChannel" and add for the value the channel name, e.g. "#dockstore-dev-testing".
+ *      Add the environment variable "hookUrl" and for the value use the webhook URL
  *
- *
- * To encrypt your secrets use the following steps:
- *
- *  1. Create or use an existing KMS Key - http://docs.aws.amazon.com/kms/latest/developerguide/create-keys.html
- *
- *  2. Click the "Enable Encryption Helpers" checkbox
- *
- *  3. Paste <SLACK_HOOK_URL> into the kmsEncryptedHookUrl environment variable and click encrypt
- *
- *  Note: You must exclude the protocol from the URL (e.g. "hooks.slack.com/services/abc123").
- *
- *  4. Give your function's role permission for the kms:Decrypt action.
- *      Example:
-
-{
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Sid": "Stmt1443036478000",
-            "Effect": "Allow",
-            "Action": [
-                "kms:Decrypt"
-            ],
-            "Resource": [
-                "<your KMS key ARN>"
-            ]
-        }
-    ]
-}
-
  */
 
-const AWS = require('aws-sdk');
 const url = require('url');
 const https = require('https');
+const AWS = require("aws-sdk");
 
-// The base-64 encoded, encrypted key (CiphertextBlob) stored in the kmsEncryptedHookUrl environment variable
-const kmsEncryptedHookUrl = process.env.kmsEncryptedHookUrl;
+// The Slack URL to send the message to
+const hookUrl = process.env.hookUrl;
 // The Slack channel to send a message to stored in the slackChannel environment variable
 const slackChannel = process.env.slackChannel;
-let hookUrl;
+const dockstoreEnvironment = process.env.dockstoreEnvironment;
+
+// Enumerate all the AWS instances in the current region
+// and find the one with the instance ID in question.
+// Then go through all the tags on that instance and
+// find the one with the key 'Name', whose value
+// is the name displayed on the console of the instance.
+function getInstanceNameAndSendMsgToSlack(targetInstanceId, messageText, processEventCallback, callback) {
+  const ec2 = new AWS.EC2();
+
+  ec2.describeInstances(function(err, result) {
+    if (err)
+      console.log(err); // Logs error message.
+    else {
+      for (var i = 0; i < result.Reservations.length; i++) {
+        var res = result.Reservations[i];
+        var instances = res.Instances;
+
+        var instance = instances.find(instance => instance.InstanceId === targetInstanceId);
+        var tagInstanceNameKey = (instance && instance.Tags.find(tag => 'Name' === tag.Key));
+        var tagInstanceName = (tagInstanceNameKey && tagInstanceNameKey.Value) || 'unknown';
+        if (tagInstanceNameKey) {
```

Comment:
If you get an error as on line 40, or you don't find tagInstanceNameKey; you're not going to send a message to Slack. I think you still should.

Also, if you're going to keep this `if`, then you can move line 48 inside it and not test for tagInstanceNameKey being set.

Reply:
fixed